### PR TITLE
[Merged by Bors] - feat(Analysis/LocallyConvex/WithSeminorms): equicontinuity criteria

### PR DESCRIPTION
**Pull request by ADedecker**
---

- [x] depends on: #5501 
- [x] depends on: #5624 
- [x] depends on: #5807

Ported from leanprover-community/mathlib#17275

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#5501
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#5501~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#5501~~
* ~~leanprover-community/mathlib4#5624~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Review by j-loreaux — COMMENTED** [review]
Looks pretty good. Minor comments.

File: Mathlib/Analysis/NormedSpace/OperatorNorm.lean [hunk 1]
```diff
@@ -1996,3 +2001,64 @@ if there is some positive constant C such that `C * ‖u‖ * ‖u‖ ≤ B u u`
...
 #align is_coercive IsCoercive
+
+section Equicontinuous
+
+variable {ι : Type _} [NontriviallyNormedField 𝕜] [NontriviallyNormedField 𝕜₂] {σ₁₂ : 𝕜 →+* 𝕜₂}
+  [RingHomIsometric σ₁₂] [SeminormedAddCommGroup E] [SeminormedAddCommGroup F]
+  [NormedSpace 𝕜 E] [NormedSpace 𝕜₂ F] (f : ι → E →SL[σ₁₂] F)
+
+/-- Equivalent characterizations for equicontinuity of a family of continuous linear maps
+between normed spaces. -/
```
[the remark this answers is not in the document]
Comment (ADedecker) [reply Mathlib/Analysis/NormedSpace/OperatorNorm.lean:2012]: Well we have both [t1Space_TFAE](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Separation.html#t1Space_TFAE) and [IsGalois.tfae](https://leanprover-community.github.io/mathlib4_docs/Mathlib/FieldTheory/Galois.html#IsGalois.tfae) so I don't think there's a clear convention yet...

File: Mathlib/Analysis/LocallyConvex/WithSeminorms.lean [hunk 2]
```diff
@@ -655,6 +656,77 @@ theorem cont_normedSpace_to_withSeminorms (E) [SeminormedAddCommGroup E] [Normed
...
+  · exact UniformEquicontinuous.equicontinuous
+  tfae_have 2 → 1
+  · exact fun H ↦ H 0
+  tfae_have 3 → 5
+  · intro H
+    have : ∀ᶠ x in 𝓝 0, ∀ k, q i (f k x) ≤ 1 := by
+      filter_upwards [Metric.equicontinuousAt_iff_right.mp (H.equicontinuous 0) 1 one_pos]
+        with x hx k
+      replace hx : dist (f k 0) (f k x) ≤ 1 := (hx k).le
+      rwa [map_zero, dist_zero_left] at hx
```
[the remark this answers is not in the document]
Comment (ADedecker) [reply Mathlib/Analysis/LocallyConvex/WithSeminorms.lean:690]: You discovered my secret: `simp` is still too scary for me, I don't even think about it when `rw` works :sweat_smile:

File: Mathlib/Analysis/NormedSpace/OperatorNorm.lean (hunk 1 above)
[the remark this answers is not in the document]
Comment (j-loreaux) [reply Mathlib/Analysis/NormedSpace/OperatorNorm.lean]: I've asked on [Zulip](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/Naming.20convention/near/383372035), but that's no reason to hold this up.

**Comment by j-loreaux** [comment]
Thanks!

bors merge

**Pull request closed without merging**